pkg/usagestats/report.go

```diff
@@ -0,0 +1,112 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package usagestats
+
+import (
+	"expvar"
+	"runtime"
+	"time"
+
+	prom "github.com/prometheus/prometheus/web/api/v1"
+
+	"github.com/grafana/mimir/pkg/util/version"
+)
+
+// Report is the JSON object sent to the stats server
+type Report struct {
+	// ClusterID is the unique Mimir cluster ID.
+	ClusterID string `json:"clusterID"`
+
+	// CreatedAt is when the cluster was created.
+	CreatedAt time.Time `json:"createdAt"`
+
+	// Interval is when the report was created (value is aligned across all replicas of the same Mimir cluster).
+	Interval time.Time `json:"interval"`
```

Comment:
This is confusing to me: why `reportAt` is `inteval`? In general, I'd never expect a property called `Interval` to be `time.Time`.

Reply:
Constructive feedback: I think this should be called `ReportedAt` (because `CreatedAt` is already taken above, although I think that `Report.CreatedAt` should indicate when _reported was created at_, and there should be `ClusterCreatedAt`, just like there's `ClusterID` and not just `ID`)

----

_I know I should have given this feedback on the design, but now I see it's clearer_.

Reply:
I can change the field name, but not the JSON name. Unfortunately we have to get stick to the way it was originally implemented by Loki (protocol shouldn't/can't change). I personally think there were few bad original design decisions, but I have to live with them.